pkg/gitprovider/bitbucket.go

```diff
@@ -508,3 +511,112 @@ type prResponseData struct {
 		} `json:"source"`
 	} `json:"values"`
 }
+
+func (g *BitbucketGitProvider) RegisterPrebuildWebhook(repo *GitRepository, endpointUrl string) (string, error) {
+	client := g.getApiClient()
+
+	hook, err := client.Repositories.Webhooks.Create(&bitbucket.WebhooksOptions{
+		Active:   true,
+		Owner:    repo.Owner,
+		Events:   []string{"repo:push"},
+		Url:      endpointUrl,
+		RepoSlug: repo.Id,
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	return hook.Uuid, nil
+}
+
+func (g *BitbucketGitProvider) GetPrebuildWebhook(repo *GitRepository, endpointUrl string) (*string, error) {
+	client := g.getApiClient()
+
+	hooks, err := client.Repositories.Webhooks.List(&bitbucket.WebhooksOptions{
+		Owner:    repo.Owner,
+		RepoSlug: repo.Id,
+		Events:   []string{"repo:push"},
+		Active:   true,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(hooks) == 0 {
+		return nil, nil
+	}
+
+	for _, hook := range hooks {
+		if hook.Url == endpointUrl {
+			return util.Pointer(hook.Uuid), nil
+		}
+	}
+
+	return nil, nil
+}
+
+func (g *BitbucketGitProvider) UnregisterPrebuildWebhook(repo *GitRepository, id string) error {
+	client := g.getApiClient()
+
+	_, err := client.Repositories.Webhooks.Delete(&bitbucket.WebhooksOptions{
+		Owner:    repo.Owner,
+		RepoSlug: repo.Id,
+		Uuid:     id,
+	})
+
+	return err
+}
+
+func (g *BitbucketGitProvider) GetCommitsRange(repo *GitRepository, owner string, initialSha string, currentSha string) (int, error) {
```

Comment:
It seems that this function is not correctly implemented according to the failing checks.
It says:
![Screenshot 2024-09-06 at 10 13 17](https://github.com/user-attachments/assets/e2c34670-23a1-47bb-8912-52a9cf3df198)
This was not an issue earlier but you must've changed something later and ended up having all these failing issues because of this. Let's revert/fix this. :)

Reply:
I guess its failing because of [this](https://github.com/daytonaio/daytona/pull/1050)